File: src/main/java/ac/dnd/server/admission/infrastructure/event/ApplicantEventListener.java
package ac.dnd.server.admission.infrastructure.event;

import org.springframework.context.event.EventListener;
import org.springframework.stereotype.Component;

import ac.dnd.server.admission.domain.event.ApplicantQueryEvent;
import lombok.extern.slf4j.Slf4j;

@Component
@Slf4j
public class ApplicantEventListener {

	@EventListener(classes = ApplicantEventListener.class)
	public void handleApplicantEvent(
		final ApplicantQueryEvent event
	) {
		log.info(
			"Applicant query event: {}",
			event
		);
	}
}

File: src/main/kotlin/ac/dnd/server/admission/infrastructure/event/ApplicantEventListener.kt
package ac.dnd.server.admission.infrastructure.event

import ac.dnd.server.admission.domain.event.ApplicantQueryEvent
import org.slf4j.LoggerFactory
import org.springframework.context.event.EventListener
import org.springframework.stereotype.Component

@Component
class ApplicantEventListener {
    private val log = LoggerFactory.getLogger(ApplicantEventListener::class.java)

    @EventListener(classes = [ApplicantQueryEvent::class])
    fun handleApplicantEvent(event: ApplicantQueryEvent) {
        log.info("Applicant query event: {}", event)
    }
}